### ISSUE TITLE
False positives from no-octal-escape in regex literals

### ISSUE THREAD

**Issue by burnnat** [eslint/eslint#582]
Consider the following javascript:

``` javascript
var regex = /([abc]) \1/g;
```

Running ESLint on this sample code yields an error from the `no-octal-escape` rule:

```
sample.js: line 1, col 12, Error - Don't use octal: '\1'. Use '\u....' instead. (no-octal-escape)

1 problem
```

However, as the [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#special-capturing-parentheses) explains, this isn't an octal escape in the regex literal, but rather a numeric backreference:

> The '`(foo)`' and '`(bar)`' in the pattern `/(foo) (bar) \1 \2/` match and remember the first two words in the string "foo bar foo bar". The `\1` and `\2` in the pattern match the string's last two words. Note that `\1`, `\2`, `\n` are used in the matching part of the regex. In the replacement part of a regex the syntax `$1`, `$2`, `$n` must be used, e.g.: `'bar foo'.replace( /(...) (...)/, '$2 $1' )`.

As such, the `no-octal-escape` rule should not consider this an error.
